src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -1547,5 +1547,24 @@ internal bool Equals(TypeSymbolWithAnnotations other)
         {
             throw ExceptionUtilities.Unreachable;
         }
+
+        public static bool Equals(TypeSymbol left, TypeSymbol right, TypeCompareKind comparison)
```

Comment:
Do we always need to provide TypeCompareKind? if so I think we should obsolete Equals(object) as well.

Reply:
Plus, is it more convenient for this to be an instance member?